src/msg/async/dpdk/DPDK.cc

```diff
@@ -644,6 +669,90 @@ DPDKQueuePair::DPDKQueuePair(CephContext *c, EventCenter *cen, DPDKDevice* dev,
     device_stat_time_fd = center->create_time_event(1000*1000, new C_handle_dev_stats(this));
 }
 
+void DPDKDevice::nic_stats_dump(Formatter *f)
+{
+  static uint64_t prev_pkts_rx[RTE_MAX_ETHPORTS];
+  static uint64_t prev_pkts_tx[RTE_MAX_ETHPORTS];
+  static uint64_t prev_cycles[RTE_MAX_ETHPORTS];
+  uint64_t diff_pkts_rx, diff_pkts_tx, diff_cycles;
+  uint64_t mpps_rx, mpps_tx;
+  struct rte_eth_stats stats;
+  size_t tx_fragments = 0;
+  size_t rx_fragments = 0;
+  size_t tx_free_cnt = 0;
+  size_t rx_free_cnt = 0;
+
+  for (auto &qp: _queues) {
+    tx_fragments += qp->perf_logger->get(l_dpdk_qp_tx_fragments);
+    rx_fragments += qp->perf_logger->get(l_dpdk_qp_rx_fragments);
+    tx_free_cnt += qp->_tx_buf_factory.ring_size();
+    rx_free_cnt += rte_mempool_avail_count(qp->_pktmbuf_pool_rx);
+  }
+  rte_eth_stats_get(_port_idx, &stats);
+  f->open_object_section("stats");
+  f->dump_unsigned("RX-packets", stats.ipackets);
+  f->dump_unsigned("DPDK RX-packets", rx_fragments);
+  f->dump_unsigned("RX-bytes", stats.ibytes);
+  f->dump_unsigned("RX-missed", stats.imissed);
+  f->dump_unsigned("RX-errors", stats.ierrors);
+
+  f->dump_unsigned("TX-packets", stats.opackets);
+  f->dump_unsigned("DPDK TX-packets", tx_fragments);
```

Comment:
i am not sure if it's valid to have a key with space(s) in it. could you use lower-case names and use underscores as delimiters like "rx_bytes" and "dpdk_tx_packets", or have dedicated sections for "rx" and "tx" respectively?